rfd/0015-agent-loading.md

```diff
@@ -0,0 +1,91 @@
+
+## Why
+
+By default `tsh login` should attempt to load the fetched certificate into the agent. If `tsh` thinks the agent does not support SSH certificates, like `gpg-agent`, Teleport should not load the certificate and instead print a warning informing the user the certificate was not loaded and how to force load.
+
+To make complex things possible a `--add-keys-to-agent` (short flag `-k`) flag shuld be added to `tsh` that can be used to control if `tsh` attempts to load a certificate into the running agent or not.
+
+This matches the [AddKeysToAgent](https://man.openbsd.org/ssh_config.5#AddKeysToAgent) option that OpenSSH supports:
+
+```
+AddKeysToAgent
+
+Specifies whether keys should be automatically added to a running ssh-agent(1). If this option is set to yes
+and a key is loaded from a file, the key and its passphrase are added to the agent with the default lifetime,
+as if by ssh-add(1). If this option is set to ask, ssh(1) will require confirmation using the SSH_ASKPASS
+program before adding a key (see ssh-add(1) for details). If this option is set to confirm, each use of the
+key must be confirmed, as if the -c option was specified to ssh-add(1). If this option is set to no, no keys
+are added to the agent. Alternately, this option may be specified as a time interval using the format
+described in the TIME FORMATS section of sshd_config(5) to specify the key's lifetime in ssh-agent(1),
+after which it will automatically be removed. The argument must be no (the default), yes, confirm (optionally
+followed by a time interval), ask or a time interval.
+```
+
+### Details
+
+The `--add-keys-to-agent` (short flag `-k`) should support following values `auto`, `yes`, `no`. The default value should be `auto`.
+
+* `auto` indicates that a best effort attempt will be made to detect the agent, and if it supports SSH certificates, load the certificate into the running agent.
+* `yes` indicates that an attemp will be made to load the certificate into the running agent.
+* `no` indicates that no attempt will be made to load the certificate into the running agent.
+
+#### Detection Algorithm
+
+The initial detection algorithm should be simple. `tsh` should read in the `$SSH_AUTH_SOCK` environment variable. If the path contains a string that indicates the agent does not support SSH certificates, like `gpg-agent`, it will not attempt to load the certificate into the agent.
+
+#### Backward Compatibility
+
+The existing certificate flags will be hidden and depreciated but will be continued to be supported forever to not break existing users workflow.
+
+The existing flags will be converted into the following values:
+
+| use-local-ssh-agent  | no-use-local-ssh-agent | add-keys-to-agent |
```

Comment:
`--use-local-ssh-agent` and `--no-use-local-ssh-agent` are the same flag: https://github.com/gravitational/teleport/blob/master/tool/tsh/tsh.go#L274-L277
You can't set `true/true` or `false/false`.